File: Backend/AstroGame.Shared/Models/Players/Perk.cs
using System;

namespace AstroGame.Shared.Models.Players
{
    public class Perk
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }

        public double BuildingSpeedMultiplier { get; set; }

        // Research
        public double EngineersResearchSpeedMultiplier { get; set; }
        public double PhysicsResearchSpeedMultiplier { get; set; }
        public double BiologicalResearchSpeedMultiplier { get; set; }

        // Resource Speeds
        public double BuildingMaterialsProductionSpeed { get; set; }
        public double ConsumablesProductionSpeedMultiplier { get; set; }
        public double ComponentsProductionSpeedMultiplier { get; set; }
        public double AlloysProductionSpeedMultiplier { get; set; }
        public double FuelsProductionSpeedMultiplier { get; set; }

        // Resource count
        public double BuildingMaterialsProductionValueMultiplier { get; set; }
        public double ConsumablesProductionValueMultiplier { get; set; }
        public double ComponentsProductionValueMultiplier { get; set; }
        public double AlloysProductionValueMultiplier { get; set; }
        public double FuelsProductionValueMultiplier { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/players/perk.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';

import 'package:json_annotation/json_annotation.dart';

part 'perk.g.dart';

@GuidConverter()
@JsonSerializable()
class Perk {
  late Guid id;
  late String title;
  late String description;

  late double buildingSpeedMultiplier;
  late double engineersResearchSpeedMultiplier;
  late double physicsResearchSpeedMultiplier;
  late double biologicalResearchSpeedMultiplier;

  late double buildingMaterialsProductionSpeed;
  late double consumablesProductionSpeedMultiplier;
  late double componentsProductionSpeedMultiplier;
  late double alloysProductionSpeedMultiplier;
  late double fuelsProductionSpeedMultiplier;

  late double buildingMaterialsProductionValueMultiplier;
  late double consumablesProductionValueMultiplier;
  late double componentsProductionValueMultiplier;
  late double alloysProductionValueMultiplier;
  late double fuelsProductionValueMultiplier;

  Perk();

  factory Perk.fromJson(Map<String, dynamic> json) => _$PerkFromJson(json);
  Map<String, dynamic> toJson() => _$PerkToJson(this);
}
